sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableAddPartitionSuite.scala

```diff
@@ -53,4 +53,19 @@ class AlterTableAddPartitionSuite
       checkAnswer(sql(s"SELECT * FROM $t"), Row(0, "abc"))
     }
   }
+
+  test("SPARK-34149: refresh cache in partition adding") {
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (part int) $defaultUsing PARTITIONED BY (part)")
```

Comment:
does this bug only exist in fully partitioned tables?

Reply:
No, it exists for any v2 tables but since our v2 table catalog is in-memory catalog, and it doesn't support `LOCATION` in `ALTER TABLE .. ADD PARTITION` (in fact it doesn't take it into account), the only way to change table is to add a partition to fully partitioned table.